pkg/plugins/layout.go

```diff
@@ -319,58 +323,69 @@ func (dev *Disk) AddPartition(label string, size uint, fileSystem string, pLabel
 		return "", err
 	}
 
-	var pDev string
 
-	for tries := 0; tries <= 5; tries++ {
-		err = dev.ReloadPartitionTable(console)
-		if err != nil {
-			log.Errorf("Failed on reloading the partition table: %v\n", err)
-			return "", err
-		}
-		pDev, err = dev.FindPartitionDevice(part.Number, console)
-		// exit if by the fifth time we cannot get the partition device
-		if err != nil && tries == 4 {
-			return "", err
-		}
-		// If no error, we got the partition, exit the loop
-		if err == nil {
-			break
-		}
-		log.Warningf("Failed to reload %s, retrying", dev.Device)
-		time.Sleep(1 * time.Second)
+	err = dev.ReloadPartitionTable(console)
+	if err != nil {
+		log.Errorf("Failed on reloading the partition table: %v\n", err)
+		return "", err
+	}
+
+	pDev, err := dev.FindPartitionDevice(part.Number, console)
+	if err != nil {
+		return "", err
 	}
 
 	mkfs := MkfsCall{part: part, customOpts: []string{}, dev: pDev}
 	return mkfs.Apply(console)
 }
 
 func (dev Disk) ReloadPartitionTable(console Console) error {
-	out, err := console.Run(fmt.Sprintf("blockdev --rereadpt %s", dev))
-	if err != nil {
-		return errors.New(fmt.Sprintf("Could not reload partition table: %s", out))
-	}
-	out, err = console.Run("sync")
-	if err != nil {
-		return errors.New(fmt.Sprintf("Could not sync: %s", out))
+	for tries := 0; tries <= partitionTries; tries++ {
+		log.Debugf("Trying to reread the partition table of %s (try number %d)", dev, tries+1)
+
+		out, err1 := console.Run(fmt.Sprintf("partprobe %s", dev))
```

Comment:
I have a doubt:

As per the fact that `partprobe `may silent return 0, when it happens, the loop in ReloadPartitionTable may break after the first try.

If the kernel side relies on the `partprobe` to trigger the probing action after device is ready (I am not sure):
   then : the loop in next "FindPartitionDevice" still has no effect.   Ideally, when fail in "`FindPartitionDevice`", force to call "`ReloadPartitionTable`", then delay, loop, make sure those actions are all executed for redundancy.

Reply:
good point, added that as well to the newest version of yip https://github.com/mudler/yip/commit/89756d49db7d4fd575a8491aa5bddc4142376060